static/app/views/dashboards/widgetCard/releaseWidgetQueries.tsx

```diff
@@ -145,267 +136,256 @@ export function requiresCustomReleaseSorting(query: WidgetQuery): boolean {
+      const useSessionAPI =
+        transformedWidget.queries[0]!.columns.includes('session.status');
+
+      let releaseCondition = '';
+      const releasesArray: string[] = [];
+      if (isCustomReleaseSorting) {
+        if (releases && releases.length === 1) {
+          releaseCondition += `release:${releases[0]!.version}`;
+          releasesArray.push(releases[0]!.version);
+        }
+        if (releases && releases.length > 1) {
+          const {releaseQueryString, releasesUsed} = getReleasesQuery(releases);
+          releaseCondition += releaseQueryString;
+          releasesArray.push(...releasesUsed);
+
+          if (!isDescending) {
+            releasesArray.reverse();
+          }
         }
       }
-    }
 
-    if (!useSessionAPI) {
-      widget.queries.forEach(query => {
-        query.conditions =
-          query.conditions + (releaseCondition === '' ? '' : ` ${releaseCondition}`);
-      });
-    }
+      if (!useSessionAPI) {
+        transformedWidget.queries.forEach(query => {
+          query.conditions =
+            query.conditions + (releaseCondition === '' ? '' : ` ${releaseCondition}`);
+        });
+      }
 
-    return widget;
-  };
+      return transformedWidget;
+    },
+    [releases]
+  );
 
-  afterFetchData = (data: SessionApiResponse) => {
-    const {widget, projects} = this.props;
-    const {releases} = this.state;
+  const afterFetchData = useCallback(
+    (data: SessionApiResponse) => {
+      const isDescending = widget.queries[0]!.orderby.startsWith('-');
 
-    const isDescending = widget.queries[0]!.orderby.startsWith('-');
+      const releasesArray: string[] = [];
+      if (requiresCustomReleaseSorting(widget.queries[0]!)) {
+        if (releases && releases.length === 1) {
+          releasesArray.push(releases[0]!.version);
+        }
+        if (releases && releases.length > 1) {
+          const {releasesUsed} = getReleasesQuery(releases);
+          releasesArray.push(...releasesUsed);
 
-    const releasesArray: string[] = [];
-    if (requiresCustomReleaseSorting(widget.queries[0]!)) {
-      if (releases && releases.length === 1) {
-        releasesArray.push(releases[0]!.version);
+          if (!isDescending) {
+            releasesArray.reverse();
+          }
+        }
       }
-      if (releases && releases.length > 1) {
-        const {releasesUsed} = getReleasesQuery(releases);
-        releasesArray.push(...releasesUsed);
 
-        if (!isDescending) {
-          releasesArray.reverse();
-        }
+      if (releasesArray.length) {
+        data.groups.sort((group1, group2) => {
+          const release1 = group1.by.release;
+          const release2 = group2.by.release;
+          // @ts-expect-error TS(2345): Argument of type 'string | number | undefined' is ... Remove this comment to see the full error message
+          return releasesArray.indexOf(release1) - releasesArray.indexOf(release2);
```

Comment:
Any chance we can get rid of this ts-expect while were here, or as a follow up. In general I'm not a bit fan of ignoring ts-errors, feels a bit hacky.